Secure hash-based password authentication

JorgeFeijoom opened the pull request:
Register, password recovery, login and profile edition hash-based

lortschi commented:
looks all fine to me

ngoyal16 commented on config/server.production.js:
```diff
@@ -32,16 +32,7 @@ module.exports = {
-		fromName: '',
-		fromAddress: ''
-	},
+	smtpServer: {},
```
smtpServer setting should be back

ngoyal16 commented on src/ajaxRouter.js:
```diff
@@ -139,57 +140,56 @@ ajaxRouter.get('/cart', (req, res, next) => {
-			.then(({ status, json }) => {
+		// update customer password after checking customer id
+		if ('id' in req.body) {
+			api.customers.update(userId, customerDraft).then(({ status, json }) => {
```
use async/await for better managment

ngoyal16 commented on src/ajaxRouter.js:
```diff
@@ -139,57 +140,56 @@ ajaxRouter.get('/cart', (req, res, next) => {
-			data.status = true;
-			data.id = AuthHeader.encodeUserLoginAuth(userId);
+		// if customer email exists send status back
+		api.customers.list(filter).then(({ status, json }) => {
```
put asyn/await before promise function else it will not wait for results

ngoyal16 commented on src/ajaxRouter.js:
```diff
@@ -305,25 +308,35 @@ ajaxRouter.post('/login', async (req, res, next) => {
+				if (out == true) {
+					customerData.token = AuthHeader.encodeUserLoginAuth(result._id);
+					customerData.authenticated = true;
+					api.customers.retrieve(result._id).then(({ status, json }) => {
```
async function when doing any call which is should be require in same thread

ngoyal16 commented on src/index.js:
```diff
@@ -33,14 +33,24 @@ app.use(express.static('public/content', STATIC_OPTIONS));
+	//res.setHeader('Access-Control-Allow-Origin', origin);
+	//}
+	//}
+	var allowedOrigins = [
```
do not hardcore cors header.. it should come to from config file.

LuBuss commented on src/ajaxRouter.js:
```diff
@@ -10,9 +10,10 @@ import AuthHeader from './lib/auth-header';
 import SettingsService from './services/settings/settings';
+import bcrypt from 'bcrypt';
 
+const saltRounds = 10;
```
Salt should be config option

JorgeFeijoom commented:
- fixed hardcore CORS header
- smtp parameters on server.production.js
- set salt rounds on server conf
- fixed async/await on api calls and better management
